libsolidity/interface/SourceReferenceFormatter.cpp

```diff
@@ -49,6 +49,21 @@ void SourceReferenceFormatter::printSourceLocation(
 	if (startLine == endLine)
 	{
 		string line = scanner.lineAtPosition(_location->start);
+
+		int locationLength = endColumn - startColumn;
+		if (locationLength > 150)
+		{
+			line = line.substr(0, startColumn) + line.substr(startColumn, 15) + "..." + line.substr(endColumn - 15, 15) + line.substr(endColumn, line.length() - endColumn);
+			endColumn = startColumn + 33;
+			locationLength = 33;
```

Comment:
Hm, having seen that, perhaps the restriction to 15 is a bit too small, since it means we drop from 149 to 33. Perhaps we should rather cut it in the middle so that the length is never larger than 150 characters?

Reply:
if it was cut in the middle and take the first half i guess, there is still chance that the half is > 150 chars. If i get the first 150 chars in the half, the total length of the line is likely to be > 150 chars, which means the text around the location is likely to be convert to "... "

Reply:
i suggest if the source location is > 150 chars, we only take the first 75 chars and append "... " as a shortened source location.

Reply:
@chriseth please check the latest PR.